client/me/purchases/manage-purchase/index.jsx

```diff
@@ -483,9 +484,14 @@ const ManagePurchase = React.createClass( {
 		const purchase = getPurchase( this.props ),
 			{ id, payment } = purchase;
 
-		if ( showEditPaymentDetails( purchase ) ) {
+		let path = paths.editCardDetails( this.props.selectedSite.slug, id );
+		if ( payment.creditCard && payment.creditCard.id ) {
```

Comment:
I think we should use `isPaidWithCreditCard` or `hasCreditCardData` from `lib/purchases` for this condition.